44.md

```diff
@@ -48,6 +49,206 @@ On its own, messages sent using this scheme have a number of important shortcomi
 Lack of forward secrecy may be partially mitigated by only sending messages to trusted relays, and asking
 relays to delete stored messages after a certain duration has elapsed.
 
+## Version 3
+
+NIP-44 version 3 improves on version 2 with the following additions: 
+- The maximum message length of 65,535 bytes is increased to 4,294,967,295 bytes
+- The encoding includes a 4th parameter to define the source pubkey used in the encryption to decouple the 
+encrypting pubkey from the event's pubkey
+- The encoding includes a 5th parameter to be used as a key derivation nonce (not to be confused with 
+the message nonce). This kd-nonce is designed to decouple Nostr's encryption from Nostr's identity, allowing 
+the creation of stable, but rotatable long-term keys for all encryptions in Nostr. 
+- The new ciphertext encoding includes a description to be shown to the user when asking for permissions 
+to decrypt the content.
+
+### Setup 
+
+Any Nostr user can publish a set of pubkeys to be replace the user's main pubkey when performing encryptions. 
+Kind `10044` lists all active pubkeys and their respective 32-byte nonce in a lowercase hex as `n` tags. 
+
+```
+{
+  "kind": 10044,
+  "tags": [
+    ["n", "<pubkey-in-lowercase-hex>", "<32-byte-nonce-in-lowercase-hex>"]
+    ...
+  ]
+}
+```
+
+To create a new pubkey, a random 32-byte kd-nonce should be generated. The private key of the item is computed by
+applying HKDF using a salt that is the concatenation of `nip88kd` in UTF-8 bytes with the kd-nonce in bytes:
+
+```
+sha256(hkdf(private_key, salt: utf8_encode('nip88kd') + <kd-nonce-byte-array>))
+```
+
+This private key doesn't need to be stored anywhere and can be cached in memory. 
+
+The pubkey is produced from the private key and then added to event kind `10044`
+
+In the absense of any `10044` event, users should encrypt using the destination's main public key.
+
+### Encryption
+
+1. Pick the destination's pubkey B
+   - 1.1 If no kind `10044` is found, use the destination's main pubkey and set the `destination-kd-nonce` to zeros
+   - 1.2 If kind `10044` is found, pick one key as pubkey and set the `destination-kd-nonce` to the nonce of that key.
+2. Pick the source's private key A 
+   - 2.1 If no kind `10044` is found, use the source's main pubkey and set the `source-kd-nonce` to zeros
+   - 2.2 If kind `10044` is found, set the `source-kd-nonce` to the nonce of that key and derive private key A
+     from the source's main private_key as `sha256(hkdf(main_private_key, salt: 'nip88kd<kd-nonce-in-hex>'))`
+   - Set `source-public-key` as the pubkey of the picked private key A.  
+3. Calculate a conversation key
+   - Execute ECDH (scalar multiplication) of public key B by private key A
```

Comment:
Indicate this is same as v2 perhaps with `(identical to v2)`  (and to subsequent sections that haven't changed)